src/snowflake/connector/cursor.py

```diff
@@ -1015,22 +1015,26 @@ def executemany(self, command, seqparams):
             self.execute(command, param, _do_reset=False)
         return self
 
-    def fetchone(self):
-        """Fetches one row."""
-        if self._prefetch_hook is not None:
-            self._prefetch_hook()
-        if self._result is None and self._result_set is not None:
-            self._result = iter(self._result_set)
-        try:
-            _next = next(self._result)
+    def _result_iterator(self):
+        """Yields the elements from _result and raises an exception when appropriate."""
+        for _next in self._result:
             if isinstance(_next, Exception):
                 Error.errorhandler_wrapper_from_ready_exception(
                     self._connection,
                     self,
                     _next,
                 )
             self._rownumber += 1
-            return _next
+            yield _next
```

Comment:
This function is an iterator over `self._result` that raises exceptions as it iterates. This behaviour is used in both `fetchone` and `__iter__` so I pulled it out to a common function here.